From 357fab10b79e49dc9b8f42dc6cf425a8124a3f40 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd@kuhls.net>
Date: Sat, 2 May 2026 16:45:38 +0200
Subject: [PATCH] Fix build with OpenSSL 4.0.0

src/base/net/tls/TlsGen.cpp: In member function 'bool
 xmrig::TlsGen::generate_x509(const char*)':
src/base/net/tls/TlsGen.cpp:118:32: error: invalid conversion from
 'const X509_name_st*' to 'X509_NAME*' {aka 'X509_name_st*'}
 [-fpermissive]
  118 |     X509_NAME_add_entry_by_txt(name, "CN", MBSTRING_ASC, reinterpret_cast<const uint8_t *>(commonName), -1, -1, 0);
      |                                ^~~~
      |                                |
      |                                const X509_name_st*

https://github.com/openssl/openssl/issues/29117

Upstream: https://github.com/xmrig/xmrig/pull/3808

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 src/base/net/tls/TlsGen.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/base/net/tls/TlsGen.cpp b/src/base/net/tls/TlsGen.cpp
index ad2e648b..59e4f0c9 100644
--- a/src/base/net/tls/TlsGen.cpp
+++ b/src/base/net/tls/TlsGen.cpp
@@ -115,7 +115,7 @@ bool xmrig::TlsGen::generate_x509(const char *commonName)
     X509_gmtime_adj(X509_get_notAfter(m_x509), 315360000L);
 
     auto name = X509_get_subject_name(m_x509);
-    X509_NAME_add_entry_by_txt(name, "CN", MBSTRING_ASC, reinterpret_cast<const uint8_t *>(commonName), -1, -1, 0);
+    X509_NAME_add_entry_by_txt((X509_NAME *) name, "CN", MBSTRING_ASC, reinterpret_cast<const uint8_t *>(commonName), -1, -1, 0);
 
     X509_set_issuer_name(m_x509, name);
 
-- 
2.47.3

